src/applications/vaos/services/appointment/transformers.v2.js

```diff
@@ -158,22 +125,76 @@ export function transformVAOSAppointment(appt) {
       isAtlas,
       atlasLocation: isAtlas ? getAtlasLocation(appt) : null,
       atlasConfirmationCode: appt.telehealth?.atlas?.confirmationCode,
+    };
+  }
+
+  let requestFields = {};
+  const appointmentType = getAppointmentType(appt);
+  if (
+    appointmentType === APPOINTMENT_TYPES.request ||
+    appointmentType === APPOINTMENT_TYPES.ccRequest
+  ) {
+    requestFields = {
+      requestedPeriod: appt.requestedPeriods,
+      // TODO: ask about created and other action dates like cancelled
+      created: null,
+      reason: PURPOSE_TEXT.find(purpose => purpose.serviceName === appt.reason)
+        ?.short,
+      preferredTimesForPhoneCall: appt.preferredTimesForPhoneCall,
+      requestVisitType: getTypeOfVisit(appt.kind),
+      practitioners: appt.practitioners,
+      // TODO: ask about service types for CC and VA requests
+      type: {
+        coding: [
+          {
+            code: appt.serviceType || null,
+            display: getTypeOfCareById(appt.serviceType)?.name,
+          },
+        ],
+      },
+      contact: {
+        telecom: appt.contact?.telecom.map(contact => ({
+          system: contact.type,
+          value: contact.value,
+        })),
+      },
+    };
+  }
+
+  return {
+    resourceType: 'Appointment',
+    id: appt.id,
+    status: appt.status,
+    // TODO Unclear what these reasons are
+    // cancelationReason: appt.cancellationReason,
+    start: start.format(),
+    // This contains the vista status for v0 appointments, but
+    // we don't have that for v2
+    description: appt.kind !== 'cc' ? 'UNKNOWN' : null,
```

Comment:
Is `UNKNOWN` a defined status? Should this field just default to null?

Reply:
We filter out appointments if their description is `null`. I wasn't sure if that was ok to remove, so I made up a status t go here so that it isn't null. I probably should name it so it's more obvious that it's VAOS related.